lib/Runtime/Library/InJavascript/Intl.js

```diff
@@ -232,10 +231,27 @@
             subTags = filtered;
         }
 
+        // As long as we are using the JS version of getExtensions on ICU, "u" will be considered an extension
+        // of a locale like "de-u-co-phonebk"
+        // Thus, we can't add the -u- ourselves here
+        const withoutSubTags = resolved;
+        if (resolved) {
+            if (subTags && getArrayLength(subTags) > 0) {
+                if (isICU) {
+                    resolved += "-";
+                } else {
+                    resolved += "-u-";
+                }
+            }
+
+            resolved += callInstanceFunc(ArrayInstanceJoin, subTags, "-");
+        } else {
+            resolved = undefined;
+        }
         return setPrototype({
-            locale: resolved ? (resolved + ((subTags && getArrayLength(subTags) > 0) ? "-u-" : "") + callInstanceFunc(ArrayInstanceJoin, subTags, "-")) : undefined,
+            locale: resolved,
             subTags: subTags,
-            localeWithoutSubtags: resolved
+            localeWithoutSubtags: withoutSubTags
```

Comment:
Can you explain this change?

Reply:
Since I moved the `resolved` logic out into its own if/else block, `resolved` here includes the subtags. `withoutSubTags` is just whatever `resolved` was before the if block runs.

Reply:
Gotcha. Thanks.